### ISSUE TITLE
DataTableRenderer: Enable/Disable default column rendering

### ISSUE THREAD

**Issue by fcjm** [primefaces/primefaces#6933]
**Description**
In Primefaces 8 it was possible to apply custom formatting when rendering the content of a column in a DataTable by e.g. checking whether column.children is empty or not. Now in Primefaces 10 there is the following Code in the DataTableRenderer.encodeCell() method preventing this:

            if (this.hasColumnDefaultRendering(table, column)) {
                Object value = UITable.createValueExprFromVarField(context, table.getVar(), column.getField()).getValue(context.getELContext());
                if (value != null) {
                    writer.writeText(value, (String)null);
                }
            } else if (column instanceof DynamicColumn) {
                column.encodeAll(context);
            } else {
                column.renderChildren(context);
            }

Basically what we were doing in our code is a similar check to hasColumnDefaultRendering() in column.renderChildren(), where we then applied our own formatting rules. But this code now will not be reached since the first "if" always catches these instances and applies the default rendering rules.

**Describe the solution you would like**
It would be nice if there was a way to enable and disable the default rendering of column content by e.g. setting a boolean on the DataTable or Column.

**Comment by Rapster** [comment]
In RC2, you should be able to override DataTable#encodeDefaultFieldCell. I'm not in favor of a new attribute for now. Also, https://github.com/primefaces/primefaces/issues/6897 might interest you

**Comment by fcjm** [comment]
Okay, thanks for the quick reply i will look into it!

**Comment by melloware** [comment]
So @Rapster what is the action here?  WontFix or is there some change you need to make to make encodeCell available in RC2?

**Comment by Rapster** [comment]
Closing for now
